### PR TITLE
Update spec for tkgs cluster corresponding to changes in API

### PR DESCRIPTION
**Pull request by Vasundharashukla**
Signed-off-by: Vasundhara Shukla <vasundharas@vmware.com>

1. **What this PR does / why we need it**:
Add new spec attributes for TKGs workload cluster.

2. **Additional information**
Also added changes to example, resource templates and documentation.

4. **Special notes for your reviewer**

### CODE REVIEW

**Review by shreyassreenivas — APPROVED** [review]

**Comment by shreyassreenivas** [comment]
Ensure these changes are tested and are working as expected

**Comment by shreyassreenivas** [comment]
Acceptance is pending

File: internal/resources/cluster/tkgservicevsphere/topology_flatten_test.go [hunk 1]
```diff
@@ -95,8 +95,9 @@ func TestFlattenTopology(t *testing.T) {
 							NodeLabels:      map[string]string{"node-key": "node-value"},
 							WorkerNodeCount: "1",
 							TkgServiceVsphere: &nodepoolmodel.VmwareTanzuManageV1alpha1ClusterNodepoolTKGServiceVsphereNodepool{
-								Class:        "test-class-spec",
-								StorageClass: "test-storage-spec",
+								Class:         "test-class-spec",
+								StorageClass:  "test-storage-spec",
+								FailureDomain: "",
```
Comment (shobha2626) [inline internal/resources/cluster/tkgservicevsphere/topology_flatten_test.go:100]: Let's have the value populated in at least some UT cases

File: internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go [hunk 2]
```diff
@@ -273,6 +273,24 @@ var distribution = &schema.Schema{
...
+				Description: "Name of the OS used for the cluster",
+				Optional:    true,
+				Computed:    true,
+			},
+			osVersionKey: {
+				Type:        schema.TypeString,
+				Description: "Version of the OS used for the cluster",
+				Optional:    true,
+				Computed:    true,
+			},
```
Comment (shobha2626) [inline internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go:293]: We can't can UT's for these new parameters?

**Review by shobha2626 — CHANGES_REQUESTED** [review]

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]
LGTM.

- Please add any required references to TMC documentation for this advanced config.
- Please update the unit tests and acceptance tests.

**Review by ishangupta-ds — APPROVED** [review]

**Review by shreyassreenivas — APPROVED** [review]

**Review by shobha2626 — APPROVED** [review]

**Pull request merged**